app/scripts/lib/config-loader.js

```diff
@@ -48,15 +60,57 @@ function (
         // did not receive the cookie.
       }
 
-      var self = this;
-      return xhr.getJSON('/config')
-          .then(function (config) {
-            self._config = config;
-            return config;
-          });
+      self._configPromise = self._xhr.getJSON('/config')
+        .fail(function (jqXHR) {
+          throw ConfigLoader.Errors.normalizeXHRError(jqXHR);
+        });
+
+      return self._configPromise;
     },
   };
 
+  var t = function (msg) {
+    return msg;
+  };
+
+  ConfigLoader.Errors = _.extend({}, Errors, {
+    ERRORS: {
+      SERVICE_UNAVAILABLE: {
+        errno: 998,
+        message: t('System unavailable, try again soon')
+      },
+      UNEXPECTED_ERROR: {
+        errno: 999,
+        message: t('Unexpected error')
+      }
+    },
+    NAMESPACE: 'config',
+
+    normalizeXHRError: function (xhr) {
+      var err;
+      var status = (xhr && xhr.status) || 0;
+      if (! status) {
+        err = this.toError('SERVICE_UNAVAILABLE');
+      } else {
+        // There are no expected errors when fetching config.
+        // All other errors are unexpected errors.
+        err = this.toError('UNEXPECTED_ERROR');
+      }
+
+      // Prepend the namespace and HTTP status to the message
+      // so at least *SOME* idea of the problem is given.
+      // The error will be reported to the error metrics.
+      err.message =
+        Strings.interpolate('%(namespace)s (%(status)s): %(message)s', {
```

Comment:
You mention in another issue using Sentry tags to capture some meta information. Could we use that here too? I'm wondering if that's a better way to handle this than encoding the namespace & status in the message.

Another thought to generalize this concept, if we can add meta information using Sentry tags, is to make [errors.js->normalizeXHRError](https://github.com/mozilla/fxa-content-server/blob/150e9db2abb684a8e36fdc09bd31fe0029fce640/app/scripts/lib/errors.js#L139) more complete. Expected errors from the auth-server, profile-server and oauth-server all return the http status in `code`. That's alright, but not really the expected place to look for it on the error. If xhr.status is 0, the status code is not captured.

What if for all XHR errors, if we add a new field to the error object, `httpStatus` or equivalent, that also captures the HTTP status code.

Whenever we capture exceptions with Raven in [lib/app-start.js](https://github.com/mozilla/fxa-content-server/blob/150e9db2abb684a8e36fdc09bd31fe0029fce640/app/scripts/lib/app-start.js#L156) and [views/base.js](https://github.com/mozilla/fxa-content-server/blob/150e9db2abb684a8e36fdc09bd31fe0029fce640/app/scripts/views/base.js#L492), we could send the http status code and error namespace as tags.